backend/app/matrix/external_url.py

```diff
@@ -0,0 +1,70 @@
+import random
+import string
+
+from app.core.background_runner import matrix_bot_runner
+
+
+class ExternalUrl:
```

Comment:
This class name feels to generic, given it's specific scope of managing the matrix "database" layer for external URLs perhaps we should make that clear in the name. I think it's especially important here as it's being passed in as a dependency in a few places.

Reply:
Done